CMakeLists.txt

```
@@ -43,29 +43,31 @@ elseif(APPLE)
     set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
 endif()
 
-function(add_cxx_flag_if_supported flag)
+function(check_if_cxx_compiler_flag_supported flag out_var)
     if(flag MATCHES "^-Wno-")
         # Compilers ignore unknown -Wno-foo flags, so look for -Wfoo instead.
         string(REPLACE "-Wno-" "-W" check_flag ${flag})
     else()
         set(check_flag ${flag})
     endif()
 
-    check_cxx_compiler_flag(${check_flag} HAVE${check_flag})
-    if(HAVE${check_flag})
+    check_cxx_compiler_flag(${check_flag} ${out_var})
+endfunction()
+
+function(add_cxx_flag_if_supported flag)
+    check_if_cxx_compiler_flag_supported(${flag} HAVE${flag})
+    if(HAVE${flag})
         add_compile_options($<$<COMPILE_LANGUAGE:CXX>:${flag}>)
     endif()
 endfunction()
 
-function(check_if_cxx_compiler_flag_supported flag out_var)
-    if(flag MATCHES "^-Wno-")
-        # Compilers ignore unknown -Wno-foo flags, so look for -Wfoo instead.
-        string(REPLACE "-Wno-" "-W" check_flag ${flag})
+function(append_source_file_compile_options file property)
+    get_source_file_property(SRC_PROPS_${file} ${file} COMPILE_OPTIONS)
+    if ("${SRC_PROPS_${file}}" STREQUAL "NOTFOUND" OR "${SRC_PROPS_${file}}" STREQUAL "")
```

```suggestion
    if (NOT OLD_OPTS)
```

The `if()` command in cmake is [wierd and magical](https://cmake.org/cmake/help/latest/command/if.html) but somehow works out well with the rest of cmake.